src/safm_api/views.py

```diff
@@ -362,8 +362,9 @@ def patch(self, request, *args, **kwargs):
     
                     # Removes the old profile picture if a new one is uploaded
                     if request.data.get('profile_picture'):
-                        path_old_picture = os.path.join(settings.MEDIA_ROOT, profile.profile_picture.name)
-                        os.remove(path_old_picture)
+                        if not profile.has_default_picture:
```

Comment:
There is a code smell from SonarCloud : better check the conditions on a single line:

```
if request.data.get('profile_picture') and not profile.has_default_picture:
```

Otherwise, everything seems fine!

Reply:
Fixed

Reply:
🔥